### DISCUSSION TITLE
Copilot not working in latest IntelliJ products EAP

### DISCUSSION THREAD

**Discussion by EdieLemoine** [community/community#34293] [Copilot Conversations] [answered]
Hello, I've just updated PhpStorm to the latest EAP version, and now GitHub Copilot is not working anymore, because of its defined version constraints. Maybe it could support "future" versions automatically... 

I guess I will roll back for now as I already miss my AI coding companion :)

**Version details:**
PhpStorm 2022.3 EAP
Build #PS-223.4884.75, built on September 22, 2022

<img width="1143" alt="Screenshot 2022-09-27 at 10 42 29" src="https://user-images.githubusercontent.com/3886637/192478116-8e1de58e-0cd0-431b-80bc-bdbf8ec68e7f.png">

Edit for the people who like to have their cake and eat it too: If you want to keep using [the new UI preview](https://blog.jetbrains.com/phpstorm/2022/09/phpstorm-2022-3-eap-is-open/#New_PhpStorm_UI_preview), you can update to 2022.3, enable the preview, then downgrade back to 2022.2. The new UI did not revert, so now I have the new UI as well as a working Copilot 😏

**Comment by CarinaSchoppe** [comment]
same for all other eap versions of jetbrains products

**Reply by EdieLemoine** [reply]
Makes sense, I've updated the title.

**Comment by azhe403** [comment]
EAP recently released, need some time for available for EAP 😐

**Comment by iamcjwilliams** [comment]
@EdieLemoine @CarinaSchoppe @azhe403 Our copilot team is investing this issue

**Reply by shyim** [reply]
Just increase in the `plugin.xml` the until-build to `223.*`

![image](https://user-images.githubusercontent.com/6224096/192862011-6e3e194f-ac0b-4ffd-993c-bdcfb2787d90.png)
Did manually for me works flawless

**Reply by CarinaSchoppe** [reply]
small description on how to do that would be kinda nice.

**Reply by shyim** [reply]
- Download from there the zip https://plugins.jetbrains.com/plugin/17718-github-copilot/versions/stable
- See here xD https://asciinema.org/a/9gw6ZPVrOBvwCYfRHkRzEgpnW

**Reply by CarinaSchoppe** [reply]
thx

**Comment by bmuskalla** [comment, marked as answer]
Preliminary support for EAP 2022.3 is available using GitHub Copilot 1.1.33.1980. We'd happy to hear any feedback if you encounter issues using Copilot on the new EAPs.

**Reply by tygas** [reply]
How about PhpStorm 2024.2 EAP?
Any hacks, cause I can not find plugin.xml in new structure

**Reply by shyim** [reply]
unpack the zip, and then it's inside lib/github-copilot-intellij-1.5.4.5592-nightly.jar

**Reply by shyim** [reply]
See https://github.com/shyim/jetbrains-github-copilot-patcher

**Comment by q42jaap** [comment]
2023.3 EAP has the same issue: no copilot support. Might I suggest that github updates the nightly build scripts to include EAP? That way any incompatibilities are found sooner by copilot users.

**Reply by shyim** [reply]
I can't find any 2023.3 EAP 🤔

**Comment by spadarthi** [comment]
2024.2 EAP  is also says copilot version not compatible

**Reply by Scott-G** [reply]
The 2024.2 EAP has been out for almost 2 weeks and STILL the CoPliot plugin does not support it.

**Reply by shyim** [reply]
See https://github.com/shyim/jetbrains-github-copilot-patcher

**Comment by wheelerlaw** [comment]
2024.3 EAP is again unsupported. Can we get a long-term solution that involves the EAP version numbers getting added proactively?

**Comment by camaeel** [comment]
Goland EAP 2025.1 is also not supported yet.

**Comment by Kajanan02** [comment]
Webstorm  2025.1 is also not supported yet.
